src/FrameManager.ts

```diff
@@ -122,18 +124,23 @@ class FrameManager {
   }
 
   private _expectedClientOrigin(): string | null {
-    if (this._frameLocation === 'about:blank') {
-      return null;
-    }
-
     try {
       const clientUrl = new URL(
         this._frameLocation,
         this._window.location.href
       );
-      return clientUrl.origin;
+
+      // Bail early if it's not an http resource.
+      // Ex: file protocol url origin behavior is browser specific
+      if (clientUrl.protocol !== 'http:' && clientUrl.protocol !== 'https:') {
+        return null;
+      }
+
+      // Assert that this origin is actually valid.
+      // Ex: about:blank, chrome:version, extensions, etc will return 'null'
+      return new URL(clientUrl.origin).origin;
```

Comment:
Looks a little weird as linting didn't like a method invocation without an assignment.  "But javascript has side effects", I yelled.

Anyway, this handles the 'null' case of about:blank and others a little more generically.